src/wallet/rpcwallet.cpp

```diff
@@ -2688,6 +2704,29 @@ static UniValue createwallet(const JSONRPCRequest& request)
     if (!wallet) {
         throw JSONRPCError(RPC_WALLET_ERROR, "Wallet creation failed.");
     }
+
+    // Encrypt the wallet if there's a passphrase
+    if (!passphrase.empty() && !(flags & WALLET_FLAG_DISABLE_PRIVATE_KEYS)) {
```

Comment:
nit: It'd be a better user experience if we error'ed out when the user provides `disable_private_keys=true` and `passphrase=something`. If they provide that input, they're expecting the wallet to be encrypted somehow. Better to error out and say "wallets without private keys cannot be encrypted" than just swallow the passphrase silently.